src/Shared/Model/Repository.cs

```diff
@@ -0,0 +1,159 @@
+        public Downloads? Downloads { get; set; }
+
+        [JsonProperty(PropertyName = "followers", NullValueHandling = NullValueHandling.Ignore)]
+        public int? FollowersCount { get; set; }
+
+        [JsonProperty(PropertyName = "forks_count", NullValueHandling = NullValueHandling.Ignore)]
+        public int? Forks { get; set; }
+
+        [JsonProperty(PropertyName = "homepage", NullValueHandling = NullValueHandling.Ignore)]
+        public string? Homepage { get; set; }
+
+        [JsonProperty(PropertyName = "is_fork", NullValueHandling = NullValueHandling.Ignore)]
+        public bool? IsFork { get; set; }
+
+        [JsonProperty(PropertyName = "language", NullValueHandling = NullValueHandling.Ignore)]
+        public string? Language { get; set; }
+
+        [JsonProperty(PropertyName = "licenses", NullValueHandling = NullValueHandling.Ignore)]
+        public List<License>? Licenses { get; set; }
+
+        [JsonProperty(PropertyName = "linked_data", NullValueHandling = NullValueHandling.Ignore)]
+        public LinkedData? LinkedData { get; set; }
+
+        [JsonProperty(PropertyName = "maintainers", NullValueHandling = NullValueHandling.Ignore)]
+        public List<User>? Maintainers { get; set; }
+
+        [JsonProperty(PropertyName = "openissues_count", NullValueHandling = NullValueHandling.Ignore)]
+        public int? OpenIssuesCount { get; set; }
+
+        [JsonProperty(PropertyName = "owner", NullValueHandling = NullValueHandling.Ignore)]
+        public User? Owner { get; set; }
+
+        [JsonProperty(PropertyName = "parent", NullValueHandling = NullValueHandling.Ignore)]
+        public string? Parent { get; set; }
+
+        [JsonProperty(PropertyName = "purl", NullValueHandling = NullValueHandling.Ignore)]
+        public string? Purl { get; set; }
+
+        [JsonProperty(PropertyName = "pushed_at", NullValueHandling = NullValueHandling.Ignore)]
+        public DateTimeOffset? PushedAt { get; set; }
+
+        [JsonProperty(PropertyName = "rank", NullValueHandling = NullValueHandling.Ignore)]
+        public double? Rank { get; set; }
+
+        [JsonProperty(PropertyName = "size", NullValueHandling = NullValueHandling.Ignore)]
+        public long? Size { get; set; }
+
+        [JsonProperty(PropertyName = "stakeholders_count", NullValueHandling = NullValueHandling.Ignore)]
+        public int? StakeholdersCount { get; set; }
+
+        [JsonProperty(PropertyName = "type", NullValueHandling = NullValueHandling.Ignore)]
+        public string? Type { get; set; }
+
+        [JsonProperty(PropertyName = "updated_at", NullValueHandling = NullValueHandling.Ignore)]
+        public DateTimeOffset? UpdatedAt { get; set; }
+
+        [JsonProperty(PropertyName = "url", NullValueHandling = NullValueHandling.Ignore)]
+        public string? Uri { get; set; }
+
+        public async Task<CSTRepository?> GetGithubRepositoryMetadata(PackageURL purl)
```

Comment:
This function is really setting the internal values of this class.  Perhaps it should be renamed `Set` or `Fetch` rather than `Get`